demoshell/main.py

```diff
@@ -137,6 +148,31 @@ def extend_text(self, style, text):
     def received_output(self, data, style):
         self.extend_text(style, data.decode('utf-8'))
 
+    def _load_config(self):
+        appname = "DemoShell"
+        appauthor = "Doug Hellman"    
+        data_dir = user_data_dir(appname,appauthor)
+        filename = "demoshell.ini"
+        filepath = os.path.join(data_dir,filename)
+
+        if (os.path.isdir(data_dir) == False ):
+            os.makedirs(data_dir)
+        
+        if (os.path.isfile(filepath) == False):
```

Comment:
The read() method of ConfigParser takes a list of filenames as argument and returns the name(s) of the files that it could load. So, instead of checking for the file up front, we could just try to read it and then if the response is that no files were read we can safely create the file. https://docs.python.org/3.5/library/configparser.html#configparser.ConfigParser.read